### DISCUSSION TITLE
Any plan to add distributed tracing support for web framework(like Gin), as Python SDK for Flask?

### DISCUSSION THREAD

**Discussion by ringsaturn** [getsentry/sentry-go#593] [Ideas]
Hello there,

In the past few years, my teams had got huge benefits from Sentry’s built-in distributed tracing support for Flask, Django, FastAPI and Celery when most of codes were written in Python.

Since we are moving to Go for performance issues, we find sentry-go has no similarly tracing support for web framework as Python SDK did.

I write a small package for Gin which seems worked: https://github.com/ringsaturn/ginsentrytracing

Hope this feature could be added to [sentry-go’s Gin support](https://github.com/getsentry/sentry-go/tree/master/gin)

I would like to open a PR for this feature if got accepted.

---

An example code:

```go
package main

import (
	"fmt"
	"net/http"
	"time"

	"github.com/getsentry/sentry-go"
	"github.com/gin-gonic/gin"
	"github.com/ringsaturn/ginsentrytracing"
)

func init() {
	err := sentry.Init(sentry.ClientOptions{
		EnableTracing:    true,
		TracesSampleRate: 1,
	})
	if err != nil {
		panic(err)
	}
}

func Foo(ctx *gin.Context) {
	span := ginsentrytracing.StartSpanFromGinContext(ctx, "slow opration")
	for i := 0; i < 5; i++ {
		childOp := span.StartChild(fmt.Sprintf("slow op: %v", i))
		time.Sleep(100 * time.Millisecond)
		childOp.Finish()
	}
	span.Finish()
	ctx.String(http.StatusOK, "bar")
	sentry.Flush(time.Second)
}

func main() {
	app := gin.Default()
	app.Use(ginsentrytracing.AttachSpan(ginsentrytracing.GetTraceAndBaggageFromRequest))

	app.GET("/foo", Foo)
	app.Run()
}
```

**Comment by smeubank** [comment]
hi @ringsaturn 

Thanks for writing in! We have recently been looking into making a sort of proposal for a roadmap to 1.0 of our Go SDK. One thing that came up which is not specifically needed for a stable 1.0 would be better framework support, Gin being one of them.

That being said, it is not planned at the moment when we could start work on it.

PR's are always welcome! @cleptric let us know if you have any specific input or guidance for a PR here

**Comment by cleptric** [comment]
@ringsaturn As Steven mentioned, we're currently looking into getting the API stable for a potential v1.0 release of the SDK.
I'm happy to accept a PR for supporting Gin though, you can take inspiration from our `net/http` integration: https://github.com/getsentry/sentry-go/blob/master/http/sentryhttp.go

**Comment by ringsaturn** [comment]
@cleptric I have opened a PR for this #600

**Comment by tonyo** [comment]
Should be addressed by https://github.com/getsentry/sentry-go/pull/644, and is now available in version v0.22.0.
